deploy/e2e/operator.yaml

```diff
@@ -0,0 +1,40 @@
+apiVersion: apps/v1
```

Comment:
Makes sense! The e2e role.yaml has additional permissions so it can create the test environment. I needed a new operator.yaml and serviceaccount.yaml because I changed the name of the role from mongodb-kubernetes-operator, however I'll look into whether we can just have role and role binding in this directory while reusing deploy/operator.yaml.

Reply:
Update: I removed operator.yaml from the new e2e directory, since we weren't using it anyway (the test deploys using deploy/operator). We do, however, need a new cluster role, cluster rolebinding, and serviceaccount to create the operator role and role binding in the test itself for the deploy. We also need these to be able to create a new namespace in the cross-namespace test.